site/docs/components/number_input.mdx

```diff
@@ -0,0 +1,181 @@
+    - You should not use steppers when large value changes are expected.
+
+## Accessibility
+- HDS number input allows the user to input the date manually without using stepper actions. Whenever you require the user to input numbers, you should also allow the manual input.
+- If steppers are used, provide descriptive aria labels for both increase and decrease actions by using `minusStepButtonAriaLabel` and `plusStepButtonAriaLabel` props.
+- If the number needs to follow a specific format, remember to specify it in the assistive text.
+- Placeholders in number inputs should be avoided. Different screen readers announce placeholders in different ways. It can be very difficult for a screen reader user to differentiate the placeholder from the actual value.
+
+## Usage and variations
+
+### Default number input
+Default HDS Number input comes without stepper. The number is always inputted manually. Using this over a text input is still helpful since it has a `type=number` to aid screen readers and mobile keyboards.
+
+When applicable, an unit can be set for the input by using the `unit` prop.
+
+<Playground>
+    <NumberInput
+        helperText="Assistive text"
+        label="Total compensation"
+        unit="€"
+        defaultValue={1000}
+        style={{ maxWidth: '320px' }}
+    />
+
+    <NumberInput
+        disabled
+        helperText="Assistive text"
+        label="Total compensation"
+        unit="€"
+        defaultValue={1000}
+        style={{ maxWidth: '320px', marginTop: 'var(--spacing-s)' }}
+    />
+</Playground>
+
+
+#### Core code example:
+```html
+<div class="hds-text-input">
+  <label for="input" class="hds-text-input__label">
+    Total compensation (€)
+    <span class="hds-text-input__required">*</span>
+  </label>
+  <div class="hds-text-input__input-wrapper">
+    <input id="input" value="1000" class="hds-text-input__input" type="number" placeholder="Placeholder" required />
+  </div>
+  <span class="hds-text-input__helper-text">Assistive text</span>
+</div>
+```
+
+#### React code example:
+```tsx
+<NumberInput
+    helperText="Assistive text"
+    label="Total compensation"
+    unit="€"
+    defaultValue={1000}
+/>
+
+<NumberInput
+    disabled
+    helperText="Assistive text"
+    label="Total compensation"
+    unit="€"
+    defaultValue={1000}
+/>
+```
+
+### Number input with steppers
+Steppers can be enabled via providing a `step` property. Use this variant when value changes are small and you can determine logical amounts for increase and decrease steps.
+
+<Playground>
+    <NumberInput
+        helperText="Assistive text"
+        label="Number of attendees"
+        minusStepButtonAriaLabel="Decrease by one"
+        plusStepButtonAriaLabel="Increase by one"
+        step={1}
+        defaultValue={5}
+        style={{ maxWidth: '320px' }}
+    />
+</Playground>
+
+
+#### Core code example:
```

Comment:
Should we delete this, or add here an example with just regular step buttons? Feels kind of desynced that the React and Core versions don't match.

Reply:
Couple of options:
1. Add an extra note that Core functionality differs from the React one.
2. Or, leave the example out but provide a Storybook link.

Reply:
Adding a text something along the lines of "Partially supported in HDS Core. See Storybook for more information.".

Reply:
Adding the extra note sounds good:

- [ ] Add an extra note that Core functionality differs from the React one.